### PR TITLE
mpu: Add helpers to enable/disable data cache for slot-2 memory

### PR DESCRIPTION
**Pull request by AntonioND**

### CODE REVIEW

**Comment by asiekierka** [comment]
Aside from the discourse on write-back, LGTM.

File: include/nds/system.h [hunk 1]
```diff
@@ -450,6 +450,14 @@ void *memCached(void *address);
 /// @return a pointer to the uncached mirror of that address.
 void *memUncached(void *address);
 
+/// Enables data cache for the DS slot-2 memory region.
```
Comment (asiekierka) [inline include/nds/system.h:453]:
```
/// Enables data cache for the DS slot-2 memory region.
///
/// Note that this is not safe to enable if you're using Slot-2 memory for
/// purposes other than external RAM, such as rumble or other I/O.
///
/// @param write_back Set as write-back. This allows writes to skip
/// the slow Slot-2 bus, at the expense of requiring a full memory flush
/// when calling peripheralSlot2DisableCache().
```

File: include/nds/system.h [hunk 2]
```diff
@@ -450,6 +450,14 @@ void *memCached(void *address);
...
+///
+/// @param write_back Set as write-back.
+void peripheralSlot2EnableCache(bool write_back);
+
+/// Disable data cache for the DS slot-2 memory region.
```
Comment (asiekierka) [inline include/nds/system.h:458]:
```
/// Disable data cache for the DS slot-2 memory region.
/// If write-back was enabled, additionally clear and flush the data cache.
///
/// Note that flushing the data cache may still be required to remove stale
/// "read" pages, such as if you write to external RAM uncached and then
/// re-enable cache.
```

Comment (AntonioND) [reply include/nds/system.h:463]: Done

File: include/nds/system.h (hunk 1 above)
Comment (AntonioND) [reply include/nds/system.h]: Done

**Pull request merged**